pkg/images/handler.go

```diff
@@ -0,0 +1,113 @@
+package images
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+
+	"github.com/containerd/containerd/content"
+	"github.com/containerd/containerd/images"
+	v1 "github.com/opencontainers/image-spec/specs-go/v1"
+	artifactspec "github.com/oras-project/artifacts-spec/specs-go/v1"
+	"oras.land/oras-go/pkg/artifact"
+	orasremotes "oras.land/oras-go/pkg/remotes"
+)
+
+// AppendArtifactsHandler will append artifacts desc to descs
+func AppendArtifactsHandler(ref, artifactType string, provider content.Provider, discoverer orasremotes.Discoverer, filters ...artifact.Filter) images.Handler {
+	return images.HandlerFunc(func(ctx context.Context, desc v1.Descriptor) ([]v1.Descriptor, error) {
+		descs := make([]v1.Descriptor, 0)
+		switch desc.MediaType {
+		case v1.MediaTypeImageManifest:
```

Comment:
Does oras support docker manifest?

Reply:
Does a registry that generates docker manifests support the referrers API? For example if in the --subject I reference a docker manifest image, do you know what happens on the server side?